alethzero/MainWin.cpp

```diff
@@ -1765,13 +1785,26 @@ void Main::on_connect_triggered()
 		ui->net->setChecked(true);
 		on_net_triggered();
 	}
-	bool ok = false;
-	QString s = QInputDialog::getItem(this, "Connect to a Network Peer", "Enter a peer to which a connection may be made:", m_servers, m_servers.count() ? rand() % m_servers.count() : 0, true, &ok);
-	if (ok && s.contains(":"))
+	
+	m_connect.setEnvironment(m_servers);
+	if (m_connect.exec() == QDialog::Accepted)
 	{
-		string host = s.section(":", 0, 0).toStdString();
-		unsigned short port = s.section(":", 1).toInt();
-		web3()->connect(host, port);
+		bool required = m_connect.required();
+		string host(m_connect.host().toUtf8().constData());
+		NodeId nodeid;
+		try
+		{
+			string nstr(m_connect.nodeId().toUtf8().constData());
+			nodeid = NodeId(fromHex(nstr));
```

Comment:
`= NodeId(fromHex(m_connect.nodeId().toStdString()))` is cleaner and doesn't require the temp var.